libraries/abstractions/ble_hal/test/ble_test_scipts/testClass.py

```diff
@@ -562,6 +568,25 @@ def get_manufacture_data(testDevice, DUT_UUID=None):
             manufacture_data = manufacture_data_dict[runTest.COMPANY_ID]
             return manufacture_data
 
+    @staticmethod
+    def get_service_data(testDevice, DUT_UUID=None):
+        service_data_dict = bleAdapter.getPropertie(
+            testDevice, "ServiceData")
+
+        # If service data doesn't exist, return None
+        if(service_data_dict is None):
+            print("No Service Data")
+            sys.stdout.flush()
+            return None
+
+        # If service data exists, return service data
+        else:
+            print("Service Data: " +
+                  str(service_data_dict.items()))
+            sys.stdout.flush()
+            service_data = service_data_dict[runTest.SERV_UUID]
```

Comment:
Add a check for runTest.SERV_UUID  key is in dictionary, otherwise test will throw an exception if its a faulty advertisement.